### PR TITLE
Integrate Jekyll SEO Tag

### PR DESCRIPTION
**Pull request by ghost**
@Phlow Before driving the new design forward I want to address a few housekeeping items to slim down the custom code being maintained. Mind taking a pass over the code?

Reference tools:
http://developers.facebook.com/tools/debug/
http://www.google.com/webmasters/tools/richsnippets

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -170,31 +181,6 @@ msapplication_tileimage:                'msapplication_tileimage.png'
...
 
-#       ______                __                __
-#      / ____/___ _________  / /_  ____  ____  / /__
-#     / /_  / __ `/ ___/ _ \/ __ \/ __ \/ __ \/ //_/
-#    / __/ / /_/ / /__/  __/ /_/ / /_/ / /_/ / ,<
-#   /_/    \__,_/\___/\___/_.___/\____/\____/_/|_|      Facebook-Optimization › http://j.mp/fb_optimize
-
-og_image:                           'http://images.phlow.de/phlow/phlow_de-logo-512x.png'
-og_locale:                          'en_EN'
-og_type:                            'website'
```
Comment (ghost) [inline _config.yml:181]: This stuff wasn't being used save for `og_locale`, which I missed with the SEO tag integration, which we can drive off the site locale instead. I can add the functionality back and open an issue against `jekyll-seo-tag`. Please let me know your thoughts.

File: _config.yml [hunk 2]
```diff
@@ -170,31 +181,6 @@ msapplication_tileimage:                'msapplication_tileimage.png'
...
-#     \__ \/ __ \/ ___/ / __ `/ /  / /|_/ / _ \/ __  / / __ `/
-#    ___/ / /_/ / /__/ / /_/ / /  / /  / /  __/ /_/ / / /_/ /
-#   /____/\____/\___/_/\__,_/_/  /_/  /_/\___/\__,_/_/\__,_/
-#
-#  Open _data/socialmedia.yml to adjust the left and right topbar navigation
-#
-
-socialmedia:
-    twitter: phlow
-    facebook: phlow.media
```
Comment (ghost) [inline _config.yml:195]: Moved social media stuff here into authors collection for better attribution for multi-author sites.

File: _drafts/gallery.md [hunk 3]
```diff
@@ -28,7 +28,7 @@ image:
 #
 # Metainformation & Customization
 #
-meta_description:
+description:
```
Comment (ghost) [inline _drafts/gallery.md:31]: The Jekyll SEO gem [uses `description`](https://github.com/jekyll/jekyll-seo-tag/#usage) to output description meta.

File: _includes/_head.html [hunk 4]
```diff
@@ -7,84 +7,71 @@
...
 
-    <link rel="stylesheet" type="text/css" href="{{ url }}/assets/css/styles_feeling_responsive.css">
+<script src="https://ajax.googleapis.com/ajax/libs/webfont/1.5.18/webfont.js"></script>
+<script>
+WebFont.load({
+google: {
+families: [ 'Lato:400,700,400italic:latin', 'Volkhov::latin' ]
+}
+});
+</script>
```
Comment (ghost) [inline _includes/_head.html:29]: This stuff wasn't actually added. I converted tabs to spaces in this file and, well, it seems like an `.editorconfig` may be in order. Thoughts?

File: _includes/_head.html [hunk 5]
```diff
@@ -7,84 +7,71 @@
...
 
-	<script src="https://ajax.googleapis.com/ajax/libs/webfont/1.5.18/webfont.js"></script>
-	<script>
-		WebFont.load({
-			google: {
-				families: [ 'Lato:400,700,400italic:latin', 'Volkhov::latin' ]
-			}
-		});
-	</script>
+{% seo %}
```
Comment (ghost) [inline _includes/_head.html:38]: This is provided by `jekyll-seo-tag` and does most of the hard work, backed by unit tests, from the RubyGem added. I added `TODO` items below for additional needs related to the integration, but didn't want to get to far before having you look this over.

File: _includes/_head.html [hunk 6]
```diff
@@ -7,84 +7,71 @@
...
-	<link rel="canonical" href="{{ url }}{{ page.url }}">
+{% assign seo_author = page.author | default: page.authors[0] | default: site.author %}
+{% if site.data.authors[seo_author] %}
+{% assign seo_author_name = site.data.authors[seo_author].name %}
+{% assign seo_author_google_plus_url = site.data.authors[seo_author].google_plus_url %}
+{% assign seo_author_facebook_user = site.data.authors[seo_author].facebook %}
+{% endif %}
 
+{% comment %} TODO: Refactor when https://github.com/jekyll/jekyll-seo-tag/issues/96 is resolved {% endcomment %}
+{% if seo_author_name %}<meta name="author" content="{{ seo_author_name }}" />{% endif %}
```
Comment (ghost) [inline _includes/_head.html:57]: This allows the site authors name to appear in Facebook cards when the author doesn't have a Facebook account. Here's how it works: https://github.com/jekyll/jekyll-seo-tag/issues/96#issuecomment-275840637

File: pages/contact.md [hunk 7]
```diff
@@ -1,7 +1,6 @@
 ---
 layout: page
 title: "Contact"
-meta_title: "Contact and use our contact form"
```
Comment (jhabdas) [inline pages/contact.md]: This was cruft

File: _config.yml [hunk 8]
```diff
@@ -140,14 +141,24 @@ defaults:
...
+twitter:
+  username: phlow
+
+# See also > _data/socialmedia.yml
+social:
+  name: Phlow
+  links:
+    - http://twitter.com/phlow
+    - http://www.mixcloud.com/phlow/
+    - https://plus.google.com/u/0/+Phlow
```
Comment (TWiStErRob) [inline _config.yml]: The public G+ url is: https://plus.google.com/+Phlow; `u/0/` is only for logged-in users. This appears in a few places.

File: _data/socialmedia.yml [hunk 9]
```diff
@@ -18,10 +18,10 @@
   class: icon-cloud
   title: "Mixe, was sonst?"
 
-- name: Phlow YouTube Google+
+- name: Google+
   url: https://plus.google.com/u/0/+Phlow
```
Comment (TWiStErRob) [inline _data/socialmedia.yml]: `u/0/`

File: _config.yml (hunk 8 above)
Comment (ghost) [reply _config.yml]: Updated two instances. Thanks for the feedback @TWiStErRob

**Comment by ghost** [comment]
@Phlow Time to take a look at this? If not I may start plowing on my fork and create a derivative work. Of course credit will be given where credit is due. 😸

File: _drafts/page_all_frontmatter.md [hunk 10]
```diff
@@ -42,18 +42,16 @@ sidebar: left
 comments: true
 breadcrumb: true
 show_meta: false
-meta_title:             # SEO: Overwrites title in <head> if needed
-meta_description:
```
Comment (ghost) [inline _drafts/page_all_frontmatter.md:46]: This stuff was removed to keep SEO meta more predictable and easier to manage using existing page description and title.

File: _includes/_head.html [hunk 11]
```diff
@@ -1,90 +1,74 @@
...
-		<link href='http://fonts.googleapis.com/css?family=Lato:400,700,400italic%7CVolkhov' rel='stylesheet' type='text/css'>
-	</noscript>
+{% comment %} TODO: Refactor when https://github.com/jekyll/jekyll-seo-tag/pull/148 lands in 2.2.0 {% endcomment %}
+{% if site.bing_webmastertools_id %}<meta name="msvalidate.01" content="{{ site.bing_webmastertools_id }}" >{% endif %}
+{% if site.yandex_site_verification %}<meta name="yandex-verification" content="{{ site.yandex_site_verification }}">{% endif %}
+{% if site.alexa_verify_id %}<meta name="alexaVerifyID" content="{{ site.alexa_verify_id }}">{% endif %}
 
+{% comment %} TODO: Refactor image usage for jekyll-seo-tag {% endcomment %}
+{% if page.image.title %}<meta property="og:image" content="{{ site.urlimg }}{{ page.image.title }}">{% endif %}
+{% if page.image.title %}<meta name="twitter:image" content="{{ site.urlimg }}{{ page.image.title }}">{% endif %}
```
Comment (ghost) [inline _includes/_head.html:47]: Future refactoring should do the following:

- Stop using `image.title` and instead use `image` or `image.path` (or `image.facebook` and `image.twitter`) which is what [Jekyll SEO Tag wants](https://github.com/jekyll/jekyll-seo-tag/#customizing-image-output).
- Allow use of both relative URLs as well as fully qualified URLs. This can be cone using the [`relative-src` include](https://github.com/CloudCannon/frisco-jekyll-template/blob/3ad14657a2c77748ab9ce1fc90d969320dc645fb/_includes/relative-src.html).
- Prevent the need to specify `header.image_fullwidth` and `image` separately. If there's a fullwidth header on the page it should automatically be used as the social share image unless overridden.

File: _includes/_pagination.html [hunk 12]
```diff
@@ -15,7 +15,7 @@
       <p>
         {% if post.image.thumb %}<a href="{{ site.url }}{{ site.baseurl }}{{ post.url }}" title="{{ post.title escape_once }}"><img src="{{ site.urlimg }}{{ post.image.thumb }}" class="alignleft" width="150" height="150" alt="{{ page.title escape_once }}"></a>{% endif %}
 
-        {% if post.meta_description %}{{ post.meta_description | strip_html | escape }}{% elsif post.teaser %}{{ post.teaser | strip_html | escape }}{% endif %}
+        {% if post.description %}{{ post.description | strip_html | escape }}{% elsif post.teaser %}{{ post.teaser | strip_html | escape }}{% endif %}
```
Comment (jhabdas) [inline _includes/_pagination.html]: Jekyll SEO tag uses `description` for SEO meta.

File: _config.yml (hunk 2 above)
Comment (Phlow) [reply _config.yml:195]: Good idea. I think sorting, making it easier and compacter is always a good idea.

File: _drafts/gallery.md (hunk 3 above)
Comment (Phlow) [reply _drafts/gallery.md:31]: I digged into the SEO-gem last week to. It adds JSON-LD into the site. which is a good think that isn't present in *Feeling Responsive*.

File: _includes/_head.html (hunk 6 above)
Comment (Phlow) [reply _includes/_head.html:57]: OK.

**Comment by Phlow** [comment]
Hey @jhabdas – I added you as a collaborator. Hey @TWiStErRob , shall I add you, too?

I never worked on GitHub together with a team. I would like to try it.

**Comment by Phlow** [comment]
Hey @jhabdas – I added you as a collaborator. Hey @TWiStErRob , shall I add you, too?

I never worked on GitHub together with a team. I would like to try it.

My suggestion. Finish v2 and than just drop it into the old repository, when we are happy with all the changes.

**Comment by Phlow** [comment]
Right now there is a conflict. I don't know how to solve it to merge your commit… Can you help me out?

**Comment by ghost** [comment]
@Phlow Thanks for making me a collaborator. I've worked on a number of teams on GitHub, both professionally and open source. There's not a lot of rhyme or reason to the way I do things as it changes based on team but I tend to value the following, from [Agile is Dead (Long Live Agility)](https://pragdave.me/blog/2014/03/04/time-to-kill-agile.html) by Dave Thomas:

> *Individuals and Interactions* over Processes and Tools
> *Working Software* over Comprehensive Documentation
> *Customer Collaboration* over Contract Negotiation, and
> *Responding to Change* over Following a Plan

I've noticed your change logs are fantastic, so please don't let the above sway you too much. The next version of Feeling Responsive will be much different and I'd like to bake in [CloudCannon support](http://cloudcannon.com/) alongside Foundation 6 from the get go given I'm building [my agency](http://comfusionllc.com) around it.

Here are a some of things I always or never do (on a team):

- Never commit straight to master
- Always work on a topic branch
- Never force push to a branch with multiple collaborators
- Always open Pull Requests before merging
- _Almost_ always PR against issues (creating them if necessary)
- Always give an :emoji: when approving a PR
- Always look for a reviewer on a PR before merging (to help avoid the [bus factor](https://en.wikipedia.org/wiki/Bus_factor))
- _Almost_ always rebase instead of produce merge commits (helps keep history clean on teams)
- Always ask before adding commits to someone else's WIP topic branches
- _Almost_ always approve PRs when the benefit is greater than the risk (or bugs introduced)

Regarding the merge conflict, no worries, I'll rebase against master and update the CHANGELOG before merging.

File: _drafts/gallery.md (hunk 3 above)
Comment (ghost) [reply _drafts/gallery.md:31]: It does, and is [currently being improved](https://github.com/jekyll/jekyll-seo-tag/pull/151) beyond what's already there.

**Comment by Phlow** [comment]
> I've noticed your change logs are fantastic

Thankyou.

> The next version of Feeling Responsive will be much different

Yes, I think so, too. To me, it's always surprising how much things change over time. When building the theme, it was the best I could do. And there is always room to improve. I am also very happy how GitHub and the developers behind Jekyll take this tool seriously.

> I'd like to bake in CloudCannon support alongside Foundation 6 from the get go given I'm building my agency around it.

Yes, let's try this. I would like to check out CloudCannon for a client. But till now I was a little to afraid to use it. But the combination of Jekyll as a generator and some handy possibility to change content easily, is the perfect thing. Especially when you compare it with WordPress.

All the Best from Germany, Moritz

**Comment by TWiStErRob** [comment]
@Phlow yes I'm in. I can help reviewing for now, but I also still have a lot of outstanding improvements from my site (and I also need to update with your improvements). Note: biggest benefit of being a collaborator is not having to fork :)

I agree with most of the points @jhabdas brought up, some comments:

 * put his list in `CONTRIBUTING.md` (I can do this if you want)
 * > Never commit straight to master  
Always open Pull Requests before merging  
Almost always PR against issues (creating them if necessary)

   that should apply to you, @Phlow, as well, it helps to make atomic improvements with well-defined scope. Having issues as a roadmap helps a lot, we can utilize milestones and labels to help filter, see [Glide repo](https://github.com/bumptech/glide/issues?page=8&q=is%3Aissue+is%3Aopen) for an exmaple. I can do labels as well, you two can figure out milestones.

 * > Always work on a topic branch  
Almost always PR against issues (creating them if necessary)

   needs a naming convention, I propose `fix/NNN_short_title` and `feature/NNN_short_title`  
`NNN` convention actually forces us to create issues. However for trivial changes it is probably a good idea to not pollute with issues. PRs are issues as well and can be labeled and milestoned.

 * > Always give an :emoji: when approving a PR  
Always look for a reviewer on a PR before merging (to help avoid the bus factor)

   We should actually use the review/reviewer feature of GitHub like we did on this issue, it's pretty good. @Phlow if you need a tour we can help. I also suggest to protect the repo with PRs "requiring review", and have @Phlow as admin so he can override if needed.

 * > Almost always rebase instead of produce merge commits (helps keep history clean on teams)

   Agree, but I also think we shouldn't amend commits in the PR's branch, it's easier to track reviews and changes. e.g. above my G+ comment is outdated and I can't review what @jhabdas actually changed in response to my comment. We can always use a squashed merge (GH supports it), or ask the contributor to rebase, squash related commits together, and force push.

**Comment by ghost** [comment]
@TWiStErRob Very glad to have your help. If you haven't tried CloudCannon yet, they offer a free trial. It'll sync easily with most all Jekyll repo in only a few seconds. I can also set up a test site for v2 as we get started. The CloudCannon team manages (and is currently overhauling) http://jekyll.tips and has an easy-to-follow set of docs on their company website covering the basics.

> I also think we shouldn't amend commits in the PR's branch

I didn't realize that. I'm onboard. Thanks for the level-up.

> needs a naming convention Re: topic branches

Not my cup of tea but happy to go with consensus.

**Comment by TWiStErRob** [comment]
Oh, one thing I forgot: I don't think it's a good idea to have a separate v2 repo. What most of the companies I worked for did is to create a release branch (`v1`)  for maintenance commits and fixes and continue working on `master` (`gh-pages` in this case). It's easier to administer IMO, especially merging between branches.

**Comment by ghost** [comment]
Closing this as there will likely be devils in the details. As @Phlow suggested this is a good v2 item when we have a chance to rethink how front matter is being used. I made a similar suggestion to @mmistakes here: https://github.com/mmistakes/minimal-mistakes/issues/324#issuecomment-282479194
